x-pack/plugins/security_solution/public/timelines/components/timeline/body/events/stateful_event.tsx

```diff
@@ -90,21 +96,50 @@ const StatefulEventComponent: React.FC<Props> = ({
 }) => {
   const trGroupRef = useRef<HTMLDivElement | null>(null);
   const dispatch = useDispatch();
+  // Store context in state rather than creating object in provider value={} to prevent re-renders caused by a new object being created
+  const [activeStatefulEventContext] = useState({ timelineID: timelineId, tabType });
   const [showNotes, setShowNotes] = useState<{ [eventId: string]: boolean }>({});
   const getTimeline = useMemo(() => timelineSelectors.getTimelineByIdSelector(), []);
-  const expandedEvent = useDeepEqualSelector(
-    (state) =>
-      (getTimeline(state, timelineId) ?? timelineDefaults).expandedEvent[
-        tabType ?? TimelineTabs.query
-      ] ?? {}
+  const expandedDetail = useDeepEqualSelector(
+    (state) => (getTimeline(state, timelineId) ?? timelineDefaults).expandedDetail ?? {}
   );
+  const hostName = useMemo(() => {
+    const hostNameArr = getMappedNonEcsValue({ data: event?.data, fieldName: 'host.name' });
+    return hostNameArr && hostNameArr.length > 0 ? hostNameArr[0] : null;
+  }, [event?.data]);
+
+  const hostIPAddresses = useMemo(() => {
+    const ipList = getMappedNonEcsValue({ data: event?.data, fieldName: 'host.ip' });
+    return ipList;
+  }, [event?.data]);
+
+  const activeTab = tabType ?? TimelineTabs.query;
+  const isDetailPanelExpanded = useMemo(() => {
```

Comment:
you do not need useMemo, it is a boolean

Reply:
Removed!